src/hardhat-deploy-ethers.ts

```diff
@@ -0,0 +1,131 @@
+/* Imports: External */
+import { Contract } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+export const registerAddress = async ({
+  hre,
+  name,
+  address,
+}): Promise<void> => {
+  // TODO: Cache these 2 across calls?
+  const { deployer } = await hre.getNamedAccounts()
+  const Lib_AddressManager = await getDeployedContract(
+    hre,
+    'Lib_AddressManager',
+    {
+      signerOrProvider: deployer,
+    }
+  )
+
+  const currentAddress = await Lib_AddressManager.getAddress(name)
+  if (address === currentAddress) {
+    console.log(
+      `✓ Not registering address for ${name} because it's already been correctly registered`
+    )
+    return
+  }
+
+  console.log(`Registering address for ${name} to ${address}...`)
+  const tx = await Lib_AddressManager.setAddress(name, address)
+  await tx.wait()
+
+  const remoteAddress = await Lib_AddressManager.getAddress(name)
+  if (remoteAddress !== address) {
+    throw new Error(
+      `\n**FATAL ERROR. THIS SHOULD NEVER HAPPEN. CHECK YOUR DEPLOYMENT.**:\n` +
+        `Call to Lib_AddressManager.setAddress(${name}) was unsuccessful.\n` +
+        `Attempted to set address to: ${address}\n` +
+        `Actual address was set to: ${remoteAddress}\n` +
+        `This could indicate a compromised deployment.`
+    )
+  }
+
+  console.log(`✓ Registered address for ${name}`)
+}
+
+export const deployAndRegister = async ({
+  hre,
+  name,
+  args,
+  contract,
+}: {
+  hre: HardhatRuntimeEnvironment
+  name: string
+  args: any[]
+  contract?: string
+}) => {
+  const { deploy } = hre.deployments
+  const { deployer } = await hre.getNamedAccounts()
+
+  const result = await deploy(name, {
+    contract,
+    from: deployer,
+    args,
+    log: true,
+  })
+
+  if (result.newlyDeployed) {
+    await registerAddress({
+      hre,
+      name,
+      address: result.address,
+    })
+  }
+}
+
+export const getDeployedContract = async (
+  hre: HardhatRuntimeEnvironment,
+  name: string,
+  options: {
+    iface?: string
+    signerOrProvider?: Signer | Provider | string
+  } = {}
+): Promise<Contract> => {
+  const deployed = await hre.deployments.get(name)
+
+  await hre.ethers.provider.waitForTransaction(deployed.receipt.transactionHash)
+
+  // Get the correct interface.
+  let iface = new hre.ethers.utils.Interface(deployed.abi)
+  if (options.iface) {
+    const factory = await hre.ethers.getContractFactory(options.iface)
+    iface = factory.interface
+  }
+
+  let signerOrProvider: Signer | Provider = hre.ethers.provider
+  if (options.signerOrProvider) {
+    if (typeof options.signerOrProvider === 'string') {
+      signerOrProvider = hre.ethers.provider.getSigner(options.signerOrProvider)
+    } else {
+      signerOrProvider = options.signerOrProvider
```

Comment:
Will this allow for the `LedgerSigner`?

Reply:
Yep, exactly

Reply:
Although realistically I still think it's better to do these deployments from hot wallets and have verification once the deployment is finished, then transfer ownership over to the multisig.